app/jobs/course/material/download_job.rb

```diff
@@ -0,0 +1,30 @@
+class Course::Material::DownloadJob < ApplicationJob
+  include TrackableJob
+
+  protected
+
+  # Performs the download service.
+  #
+  # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+  #   downloaded.
+  # @param [String] filename The name of the zip file.
```

Comment:
To make sure the user downloaded file is `filename.zip`.

Actually I can use the folder name when the folder is the only content, but when user selectively downloads the contents of the folder, there is no way to know the filename.